### PR TITLE
server: refine tiflash fallback testcase

### PR DESCRIPTION
**Pull request by xuyifangreeneyes**
### What problem does this PR solve?

Problem Summary: 
Current tiflash fallback testcases inject failure into an improper place.

### What is changed and how it works?

Use unistore to simulate tiflash fallback scenario.

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:
- Need to cherry-pick to the release branch

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

- refine tiflash fallback testcase

### CODE REVIEW

File: server/conn_test.go [hunk 1]
```diff
@@ -753,40 +753,66 @@ func (ts *ConnTestSuite) TestTiFlashFallback(c *C) {
...
+	c.Assert(failpoint.Disable("github.com/pingcap/tidb/store/mockstore/unistore/BatchCopRpcErrtiflash0"), IsNil)
+
+	// test that TiDB would not retry if the first execution already sends data to client
+	c.Assert(failpoint.Enable("github.com/pingcap/tidb/server/secondNextErr", "return(true)"), IsNil)
+	tk.MustExec("set @@tidb_enable_tiflash_fallback_tikv=1")
+	c.Assert(cc.handleQuery(ctx, "select * from t t1 join t t2 on t1.a = t2.a"), NotNil)
+	c.Assert(failpoint.Disable("github.com/pingcap/tidb/server/secondNextErr"), IsNil)
+
+	// simple TiFlash query (unary + non-streaming)
+	tk.MustExec("set @@tidb_allow_batch_cop=0; set @@tidb_allow_mpp=0; set @@tidb_enable_streaming=0;")
```
Comment (hanfei1991) [inline server/conn_test.go:782]: As I know, the enable_streaming will not affect tiflash.

**Review by xiongjiwei — APPROVED** [review]

**Review by winoros — APPROVED** [review]

**Pull request merged**